xql/main.py

```diff
@@ -155,30 +155,76 @@ def apply_order_by(fields: t.List[str], ds: xr.Dataset) -> xr.Dataset:
     return ordered_ds
 
 
-def apply_group_by(fields: t.List[str], ds: xr.Dataset, agg_funcs: t.Dict[str, str]) -> xr.Dataset:
+def aggregate_variables(agg_funcs: t.List[t.Dict[str, str]],
+                        ds: xr.Dataset,
+                        time_fields: t.List[str],
+                        coords_to_squeeze: t.List[str]) -> xr.Dataset:
+    """
+    Aggregate variables in an xarray dataset based on aggregation functions.
+
+    Args:
+        agg_funcs (List[Dict[str, str]]): List of dictionaries specifying aggregation functions for variables.
+        ds (xr.Dataset): The input xarray dataset.
+        time_fields (List[str]): List of time fields to consider for time-based grouping.
+        coords_to_squeeze (List[str]): List of coordinates to be squeezed during aggregation.
+
+    Returns:
+        xr.Dataset: The aggregated xarray dataset.
+    """
+    agg_dataset = xr.Dataset(coords=ds.coords, attrs=ds.attrs)
+
+    # Aggregate based on time fields
+    if len(time_fields):
+        agg_dataset = agg_dataset.groupby(ds['time'].dt.strftime(timestamp_formats[time_fields[0]]))
+        agg_dataset = apply_aggregation(agg_dataset, 'avg', None)
+        agg_dataset = agg_dataset.rename({"strftime": time_fields[0]})
```

Comment:
ideally we should create  a blank dataset some other way. but that can be done in later PR.

Reply:
Ok. Will do in future PR.